nemo/collections/nlp/modules/common/huggingface/huggingface_decoder.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from hydra.utils import instantiate
+from transformers import AutoConfig, AutoModel
+
+from nemo.collections.nlp.modules.common.decoder_module import DecoderModule
+from nemo.collections.nlp.modules.common.huggingface.huggingface_utils import get_huggingface_pretrained_lm_models_list
+from nemo.utils import logging
+
+
+class HuggingFaceDecoderModule(DecoderModule):
+    """ Class for using HuggingFace encoders in NeMo NLP."""
```

Comment:
typo: encoders->decoders

Reply:
I'm wondering, should we restrict the kinds of HF model which can be decoders?
For example, not quite sure how BERTs would be used as decoders. GPT, however, can be used as decoders.

Reply:
I'm not sure yet how much we want to restrict, but I think we can look at it from different levels:

- Advise in documentation and docstrings
- Add warnings based on bad choices
- Raise an error if a bad choice is made

My thinking was that encoders/decoders with the correct neural types will be able to be plugged in and at least training will run.